vlog4j-core/src/test/java/org/semanticweb/vlog4j/core/reasoner/ReasonerTimeoutTest.java

```diff
@@ -0,0 +1,99 @@
+package org.semanticweb.vlog4j.core.reasoner;
+
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeAtom;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeConstant;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makePredicate;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeRule;
+import static org.semanticweb.vlog4j.core.model.implementation.Expressions.makeVariable;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.semanticweb.vlog4j.core.model.api.Atom;
+import org.semanticweb.vlog4j.core.model.api.Predicate;
+import org.semanticweb.vlog4j.core.model.api.Rule;
+import org.semanticweb.vlog4j.core.model.api.Variable;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.EdbIdbSeparationException;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.IncompatiblePredicateArityException;
+import org.semanticweb.vlog4j.core.reasoner.exceptions.ReasonerStateException;
+
+/**
+ * Test case ensuring {@link Reasoner#setReasoningTimeout(Integer)} works as expected and terminates reasoning after the given {@link #timeout}.
+ * Results are accepted within one second to account for setup and tear down of reasoning resources.
+ * @author Adrian Bielefeldt
+ *
+ */
+public class ReasonerTimeoutTest {
+	
+	/**
+	 * The timeout after which reasoning should be completed in seconds.
+	 */
+	static int timeout = 1;
```

Comment:
what is the reason for which these fields are package protected? I would make fields private until we need to use them in other classes.

Reply:
All class variables are now private.